ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -456,7 +465,14 @@ std::tuple<ttnn::Shape, ttnn::MemoryConfig, bool> get_conv_padded_input_shape_an
         uint32_t input_num_cores_nhw = get_num_cores_nhw_from_parallel_config(parallel_config);
         // TT_ASSERT(input_tensor.get_legacy_shape() == input_tensor.get_shape());
         uint32_t tensor_height = input_tensor.get_shape()[0] * input_tensor.get_shape()[1] * input_tensor.get_shape()[2];
-        uint32_t input_tensor_height_snapped_to_tile = (shard_layout == TensorMemoryLayout::WIDTH_SHARDED)? tensor_height : tt::round_up(tensor_height, input_num_cores_nhw * 32);
+        uint32_t input_tensor_height_snapped_to_tile;
+        if(conv_config.shard_layout == TensorMemoryLayout::WIDTH_SHARDED) {
```

Comment:
you can simplify this if-else-else because width sharding case has `input_num_cores_nhw = 1`